первый этап первого спринта

yakovlevyaroslav opened the pull request.

yakovlevyaroslav committed:
первый этап первого спринта

gennady-bars commented on src/components/app/App.jsx:
```diff
@@ -0,0 +1,23 @@
+  render() {
+    return (
+      <>
+        <AppHeader></AppHeader>
```
**Можно лучше**

Можно использовать **самозакрывающиеся** теги, если нет `children` (нет ничего между тегов `><`).

yakovlevyaroslav replied:
Поправил, спасибо. В других местах проекта так и делал, а в App.jsx забыл поменять.

gennady-bars commented on src/components/app/App.jsx:
```diff
@@ -0,0 +1,23 @@
+import appStyles from './app.module.css';
+import indexStyles from '../../index.module.css'
+
+class App extends React.Component {
```
**Можно лучше**

 **классовые компоненты** редко используются в современных проектах. Даже в рассвет классов в реакте они не использовались для компонентов, если в них нет стейта: `state`. Лучше всегда использовать функциональные + хуки

yakovlevyaroslav replied:
Насколько я понял это промежуточная проверка, все на классовые компоненты я перепишу к воскресенью, это обязательно)

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

К сожалению, работу придется отклонить без потери итерации

* Проект не запускается, так как нет папки `public` https://disk.yandex.ru/i/EsI8tPSAs5OHRg 

Еще заметил ошибки:

* Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Это затрудняет управление стилями. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

**Можно лучше**

* Можно использовать **самозакрывающиеся** теги, если нет `children` (нет ничего между тегов `><`).
*  **классовые компоненты** редко используются в современных проектах. Даже в рассвет классов в реакте они не использовались для компонентов, если в них нет стейта: `state`. Лучше всегда использовать функциональные + хуки

yakovlevyaroslav committed:
убрал из .gitignore /public и запушил ее

yakovlevyaroslav committed:
поменял теги компоненов в App.jsx на самозакрывающиеся, так как в них нет содержимого

yakovlevyaroslav committed:
поменял инлайновые стили на классы css

yakovlevyaroslav commented:
> * Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Это затрудняет управление стилями. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

Это тоже поправил

yakovlevyaroslav commented:
> * Проект не запускается, так как нет папки `public` https://disk.yandex.ru/i/EsI8tPSAs5OHRg

Поправил, честно говоря думал папка создается автоматически при запуске npm start, но потом убедился, что нет. Убрал ее из .gitignore

gennady-bars commented on src/components/burger-ingredients/BurgerIngredients.jsx:
```diff
@@ -0,0 +1,133 @@
+    const buns = burgers
+      .filter(data => data.type === 'bun')
+      .map(data => (
+        <li key={data._id} className={`${burgerIngredientsStyles.burger_ingredients__item} p-4`}>
```
`key={data._id}`

Отлично, что используете атрибут `key`, и отлично, что для него используете уникальный `_id`, а не индекс массива.

[Почему индекс не нужно использовать, можно прочитать тут]( https://habr.com/ru/companies/hh/articles/352150/)

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:



* Отлично, что используете атрибут `key`, и отлично, что для него используете уникальный `_id`, а не индекс массива.


но есть некоторые недочеты:


* Скрин https://disk.yandex.ru/i/JERR7d37RJMEew горизонтальный скролл не требуется в списке ингредиентов. Сам блок с табами не должен сжиматься больше, чем 2 ингредиента по макету и так
* Собранный заказ (бургер) должен отображаться справа от списка ингредиентов, а  не внизу по ними https://disk.yandex.ru/i/uvFekCUTB7xZNA
* В верхней булке нужно добавлять текст `верх`, а в нижней - `низ`. Посмотрите в фигме, пожалуйста
* Верхняя и нижняя булки должны быть всегда **одинаковыми** по заданию и у них должна быть иконка `Замочек`, так как их нельзя будет удалять (только заменять перетаскиванием)
* Верхняя и нижняя булки должны **отличаться** по форме. Это делается с помощью атрибутов `type`

yakovlevyaroslav committed:
Внес правки по ревью:
- оставил в стилять только вертикальный скролл
- сделал перестройку интерфейся от 980px, а не от 1200px, чтобы он дольше оставался горизонтальным
- добавил булкам в компоненте BurgerConsctructor type, надписи верх и низ, и сделал так чтобы булка была одинаковая для верха и низа

gennady-bars reviewed (commented):
1й этап принят. Теперь нужен 2й с попапами


* Лишние сролл-бары тоже нужно удалить. Только один общий скролл от табов вниз (до низа экрана)
